taxonomy-editor-frontend/src/pages/project/search/FiltersArea.tsx

```diff
@@ -0,0 +1,212 @@
+  "&.Mui-checked": {
+    color: "#341100",
+  },
+};
+
+type FiltersAreaType = {
+  setCurrentPage: Dispatch<SetStateAction<number>>;
+  setQ: Dispatch<SetStateAction<string>>;
+  filters: EntryNodeSearchResult["filters"];
+};
+
+export const FiltersArea = ({
+  setCurrentPage,
+  setQ,
+  filters,
+}: FiltersAreaType) => {
+  const [isRootNodesChecked, setIsRootNodesChecked] = useState<boolean>(true);
+  const [taxonomyScope, setTaxonomyScope] = useState<string>("");
+  const [chosenLanguagesCodes, setChosenLanguagesCodes] = useState<string[]>(
+    []
+  );
+  const [withoutChosenLanguagesCodes, setWithoutChosenLanguagesCodes] =
+    useState<string[]>([]);
+  const [parentId, setParentId] = useState<string>("");
+  const [childId, setChildId] = useState<string>("");
+  const [ancestorId, setAncestorId] = useState<string>("");
+  const [descendantId, setDescendantId] = useState<string>("");
+
+  const initializeFilters = (): {
+    isRootNodesChecked: boolean;
+    taxonomyScopeMode: string; //"in" -> in taxonomy, "out" -> outside taxonomy, "" -> filter not selected
+    chosenLanguagesCodes: string[];
+    withoutChosenLanguagesCodes: string[];
+  } => {
+    return {
+      isRootNodesChecked: false,
+      taxonomyScopeMode: "",
+      chosenLanguagesCodes: [],
+      withoutChosenLanguagesCodes: [],
+    };
+  };
+
+  const updateFiltersStates = useCallback((updatedFilters) => {
+    const filtersStates = initializeFilters();
+    for (const filter of updatedFilters) {
+      switch (filter.filterType) {
+        case "is":
+          switch (filter.filterValue) {
+            case "root":
+              filtersStates.isRootNodesChecked = true;
+              break;
+            case "external":
+              filtersStates.taxonomyScopeMode = "out";
+              break;
+            case "not:external":
+              filtersStates.taxonomyScopeMode = "in";
+              break;
+          }
+          break;
+        case "language":
+          if (filter.negated) {
+            filtersStates.withoutChosenLanguagesCodes.push(
+              filter.filterValue.replace("not:", "")
+            );
+          } else {
+            filtersStates.chosenLanguagesCodes.push(filter.filterValue);
+          }
+          break;
+        default:
+          break;
+      }
+    }
+    setIsRootNodesChecked(filtersStates.isRootNodesChecked);
+    setTaxonomyScope(filtersStates.taxonomyScopeMode);
+    setChosenLanguagesCodes(filtersStates.chosenLanguagesCodes);
+    setWithoutChosenLanguagesCodes(filtersStates.withoutChosenLanguagesCodes);
+  }, []);
+
+  useEffect(() => {
+    updateFiltersStates(filters);
+  }, [filters, updateFiltersStates]);
+
+  const handleCheckbox = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    filterKey: string,
+    isChecked: boolean,
+    setChecked: React.Dispatch<React.SetStateAction<boolean>>
+  ) => {
+    setCurrentPage(1);
+    if (!isChecked) {
+      setQ((prevQ) => prevQ + ` ${filterKey}`);
```

Comment:
```suggestion
      setQ((prevQ) => `${filterKey} prevQ`);
```
I think it is better in most cases because the user is writing on the right side of the input field